.github/workflows/compiler-discovery.yml

```diff
@@ -28,15 +28,21 @@ jobs:
     steps:
     - name: Update infra
       run: cd /home/ubuntu/infra && git pull
+      shell: bash
```

Comment:
Good spot. Is this the magic to get actual failure propagation?

Reply:
I don't think it matters, this really only matters in case you do a lot of scriptlines in 1 step (I think) (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell - it adds `-norc -eo pipefail` to the bash command that runs the script).

It already propagates with the changes I made in the ce python script (the one with the empty exception handling)

So I can just remove the shell bits and it should still work

Reply:
Ok cool: I think I'd prefer not adding the `shell:` unless we need it, just to reduce noise, though if it surprises us again we can just accept the extra lines.